src/index.ts

```diff
@@ -26,21 +49,32 @@ export async function runServer(): Promise<void> {
 
 export async function main(...args: string[]): Promise<void> {
 	const opts = getOptions(...args);
+	logger.info('Main function started with options:', { once: opts.once, autoDiscover: opts.autoDiscover });
 
 	if (opts.once) {
+		logger.info('Running in printOnce mode...');
 		await printOnce();
+		logger.info('printOnce completed, forcing exit...');
+		process.exit(0); // Force exit immediately after printOnce
```

Comment:
I'd move all `process.exit` related handling to a single place, e.g. `return opts.once` from `main` and use that on line 77 to either wait for timeout or `process.exit(exitCode)` without timeout - that way we also get common error handling for free